src/components/editor/ImportResourceTemplate.jsx

```diff
@@ -43,80 +44,102 @@ class ImportResourceTemplate extends Component {
+    } catch(error) {
+      console.log('received error with create resource request:')
+      console.dir(error)
+      const createErrors = this.state.createResourceError.slice(0)
+      createErrors.push(error.response)
+      console.log('pushing to createResourceError array:')
+      console.dir(createErrors)
+      this.setState({ createResourceError: createErrors })
+      return error.response
     }
-    return response
-  }).catch(() => {})
-
-  updateResourcePromise = (content, group) => new Promise(async (resolve) => {
-    resolve(
-      await instance.updateResourceWithHttpInfo(group, content.id, content, { contentType: 'application/json' }).catch((error) => {
-        this.setState({
-          updateResourceError: error
-        })
-      })
-    )
-  }).then(response => {
-
-    return response
-  }).catch(() => {})
-
-  fulfillCreateResourcePromise = async (promise) => {
-    await promise.then(result => {
-      const joinedMessages = this.state.message.slice(0)
-
-      if(result.response.statusText !== 'No Content') {
-        this.setState({tempState: `${result.response.statusText} ${result.response.headers.location}`})
-        joinedMessages.push(this.state.tempState)
-        this.setState({message: joinedMessages})
-      }
-
-    }).catch(() => {
-      if (this.state.createResourceError[0].statusText !== 'Conflict') {
-        const msg = 'The sinopia server is not accepting the request for this resource.'
-        this.state.updateResourceError ? this.setState({message: [`${msg}: ${this.state.updateResourceError}`]}) : this.setState({message: [msg]})
-      }
-    })
+    console.log('received non-error response to create resource request:')
+    console.dir(response)
+    return response.response
   }
 
-  updateStateForResourceError = (error) => {
-    if(_.get(error, 'response.statusText') === 'Conflict') {
-      const response = error.response
-      const joinedConflicts = this.state.createResourceError.slice(0)
-      this.setState({tempConflictError: response})
-      joinedConflicts.push(this.state.tempConflictError)
-      this.setState({createResourceError: joinedConflicts})
+  updateResource = async (content, group) => {
+    let response
+    try {
+      response = await instance.updateResourceWithHttpInfo(group, content.id, content, { contentType: 'application/json' })
+    } catch(error) {
+      console.log('received error with update resource request:')
+      console.dir(error)
+      this.setState({ updateResourceError: error })
+      return error.response
     }
+    console.log('received non-error response to update resource request:')
+    console.dir(response)
+    return response.response
   }
 
-  updateAllResources = (rts, group) => {
-    return Promise.all(rts.map(async rt =>
-      await this.fulfillCreateResourcePromise(this.updateResourcePromise(rt, group))
-    ))
+  handleResponse = (response, operation, profileCount) => {
```

Comment:
consider renaming to handleTrellisResponse?  or is that too long?

Reply:
consider not using "handle" because we use that for things like "handleClick" ... maybe `processTrellisResponse` ??

"It sets the state for the trellis response" is a description

`updateStatePerTrellisResponse`   `setStatePerTrellis`

Reply:
Done in my local branch.